src/vm/environment.ts

```diff
@@ -8,15 +8,9 @@ import { JDEventSource } from "../jdom/eventsource"
 import { CHANGE, EVENT } from "../jdom/constants"
 import { jdpack, PackedValues } from "../jdom/pack"
 
-export async function refreshRegistersAsync(registers: SMap<JDRegister>) {
-    for (const k in registers) {
-        const register = registers[k]
-        let retry = 0
-        let val: any = undefined
-        do {
-            await register.refresh()
-            val = register.unpackedValue?.[0]
-        } while (val === undefined && retry++ < 2)
+export class VMError {
```

Comment:
use this pattern: https://github.com/microsoft/jacdac-ts/blob/main/src/jdom/error.ts